packages/polaris-viz/src/components/DonutChart/components/LegendValues/LegendValues.tsx

```diff
@@ -83,9 +84,53 @@ export function LegendValues({
     return Math.max(maxWidth, trendIndicatorWidth);
   }, 0);
 
+  const longestLegendNameWidth = useMemo(() => {
+    return legendData.reduce((previous, current) => {
+      const estimatedLegendNameWidth = estimateStringWidth(
+        `${seriesNameFormatter(`${current.name || ''}`)}`,
+        characterWidths,
+      );
+
+      if (estimatedLegendNameWidth > previous) {
+        return estimatedLegendNameWidth;
+      }
+
+      return previous;
+    }, 0);
+  }, [legendData, seriesNameFormatter, characterWidths]);
+
+  const longestLegendValueWidth = useMemo(() => {
+    return legendData.reduce((previous, current) => {
+      const estimatedLegendValueWidth = estimateStringWidth(
+        `${labelFormatter(`${current.value || ''}`)}`,
+        characterWidths,
+      );
+
+      if (estimatedLegendValueWidth > previous) {
+        return estimatedLegendValueWidth;
+      }
+
+      return previous;
+    }, 0);
+  }, [legendData, labelFormatter, characterWidths]);
+
+  const legendTableMaxWidth = useMemo(
+    () =>
+      longestLegendNameWidth +
+      longestLegendValueWidth +
+      maxTrendIndicatorWidth +
+      TABLE_LEGEND_PADDING,
+    [longestLegendNameWidth, longestLegendValueWidth, maxTrendIndicatorWidth],
+  );
+
   return (
     <React.Fragment>
-      <table className={styles.Table}>
+      <table
+        className={styles.Table}
+        style={{
+          maxWidth: legendTableMaxWidth,
```

Comment:
In theory, wouldn't we want this to be a `minWidth` if the problem is the table is not wide enough for the legend text?